java/statistical-query/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/MainActivity.java

```diff
@@ -0,0 +1,171 @@
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.statisticalquery;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import android.os.Bundle;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.FeatureTable;
+import com.esri.arcgisruntime.data.QueryParameters;
+import com.esri.arcgisruntime.data.ServiceFeatureTable;
+import com.esri.arcgisruntime.data.StatisticDefinition;
+import com.esri.arcgisruntime.data.StatisticRecord;
+import com.esri.arcgisruntime.data.StatisticType;
+import com.esri.arcgisruntime.data.StatisticsQueryParameters;
+import com.esri.arcgisruntime.data.StatisticsQueryResult;
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class MainActivity extends AppCompatActivity {
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private MapView mMapView;
+  private FeatureTable mWorldCitiesTable;
+
+  private CheckBox mCurrentExtentCheckbox;
+  private CheckBox mGreater5mCheckbox;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // inflate ui elements
+    mMapView = findViewById(R.id.mapView);
+    mCurrentExtentCheckbox = findViewById(R.id.currentExtentCheckBox);
+    mGreater5mCheckbox = findViewById(R.id.greater5mCheckBox);
+    Button getStatisticsButton = findViewById(R.id.getStatisticsButton);
+
+    // create a new Map with the world streets vector basemap
+    ArcGISMap map = new ArcGISMap(Basemap.createStreetsVector());
+
+    // create feature table using the world cities URI
+    mWorldCitiesTable = new ServiceFeatureTable(getString(R.string.world_cities_service));
+
+    // create a new feature layer to display features in the world cities table
+    FeatureLayer worldCitiesLayer = new FeatureLayer(mWorldCitiesTable);
+
+    // add the world cities layer to the map
+    map.getOperationalLayers().add(worldCitiesLayer);
+
+    // assign the map to the MapView
+    mMapView.setMap(map);
+
+    // add click listener to get statistics button
+    getStatisticsButton.setOnClickListener(view -> getStatistics());
+  }
+
+  private void getStatistics() {
+
+    // create definitions for each statistic to calculate
```

Comment:
This section of code runs every time the user hits the query button, but it's exactly the same each time (setting up statistic definitions). Any reason why the `List<StatisticDefinition>` is not created once and stored?